addons/sale_loyalty/models/sale_order.py

```diff
@@ -454,7 +454,7 @@ def _get_program_domain(self):
         self.ensure_one()
         today = fields.Date.context_today(self)
         return [('active', '=', True), ('sale_ok', '=', True),
-                ('company_id', 'in', (self.company_id.id, False)),
+                *self.env['loyalty.program']._check_company_domain([self.company_id.id, self.company_id.parent_id.id]),
```

Comment:
Hello @anko-odoo ! 

I thought it was redundant to include `False` as an argument for `_check_company_domain` when it also includes and `or` clause for `False` as shown here: https://github.com/odoo/odoo/blob/d0b0ff849c1df7d563bfb007b84efd0b4c1659ea/odoo/models.py#L4030

Unless you think it is best to explicitly include False into the arguments for future-proofing?

Reply:
Yes, but I mean why use _check_company_domain in this version of fix and not the other?

Reply:
Oh, the reasoning behind using _check_company_domain was because @Demesmaeker mentioned that it was introduced starting saas-16.4 and suggested that we adapt the method in this version. See conversation here: https://github.com/odoo/odoo/pull/178135#issuecomment-2385671831